src/openalea/rsml/io.py

```diff
@@ -48,7 +48,18 @@ def parse(self, filename, debug=False):
         root = doc.getroot()
         # recursive call of the functions to add neww plants/root axis to the MTG
         self.dispatch(root)
-
+        
+        # if some functions are defined in the MTG properties but not in metadata, add them
+        graph = self._g 
+        if graph.graph_properties().get('metadata', {}).get('functions') is None: 
```

Comment:
Get just ones graph_properties rather than calling each time.
Code will be simpler, nicer.

```python
props = graph.graph_properties()

Reply:
Or directly for metadata

```python
metadata = graph.graph_properties().set_default('metadata', {})
```